cli/compose/convert/volume.go

```diff
@@ -22,43 +22,40 @@ func Volumes(serviceVolumes []composetypes.ServiceVolumeConfig, stackVolumes vol
 	return mounts, nil
 }
 
-func convertVolumeToMount(
-	volume composetypes.ServiceVolumeConfig,
-	stackVolumes volumes,
-	namespace Namespace,
-) (mount.Mount, error) {
+func createMountFromVolume(volume composetypes.ServiceVolumeConfig) mount.Mount {
 	result := mount.Mount{
 		Type:        mount.Type(volume.Type),
-		Source:      volume.Source,
 		Target:      volume.Target,
 		ReadOnly:    volume.ReadOnly,
 		Consistency: mount.Consistency(volume.Consistency),
 	}
-
-	// Anonymous volumes
-	if volume.Source == "" {
-		return result, nil
-	}
-	if volume.Type == "volume" && volume.Bind != nil {
-		return result, errors.New("bind options are incompatible with type volume")
-	}
-	if volume.Type == "bind" && volume.Volume != nil {
-		return result, errors.New("volume options are incompatible with type bind")
+	if volume.Type != "tmpfs" {
+		result.Source = volume.Source
```

Comment:
This isn't really necessary because `handleTmpfsToMount()` will error out in this case. I would remove this so we don't have to check it in two places.